enhancements/multi-arch/heterogeneous-architecture.md

```diff
@@ -0,0 +1,348 @@
+
+### API Extensions
+
+Upgrades to a heterogeneous payload will require an additional field either in the Infrastructure or the ClusterVersion resource to indicate that
+the user wants to upgrade to a heterogeneous payload.
+
+### Implementation Details/Notes/Constraints
+
+Operators supporting multiple architectures cannot stop supporting a given architecture since it would break heterogeneous clusters which are
+trying to upgrade.
+
+### Risks and Mitigations
+
+Heterogeneous payloads will only ship if the corresponding homogeneous payloads for all architectures can ship and are not broken due to a bug in a
+specific architecture.
+
+## Design Details
+
+### RHCOS Support for ARM Azure
+
+The RHCOS pipeline will need to generate Azure disk images for ARM. However as of today, RHEL does not support hyper-v for ARM and that support is
+landing in RHEL8.6. Hence 4.11 builds of RHCOS for ARM Azure will be based on RHEL8.6. It is TBD whether a separate pipeline with early access RHCOS
+builds will be needed as RHEL8.6 would not have GA'ed before 4.11.
+
+### OCP Build & Release
+
+Phase 0 will involve building and publishing of [nightly builds][OCPPLAN-7640]. The release payload will be a manifestlist-based image with the
+component images being manifestlist references as well. All 4 architectures - x86, ARM, Power and Z will be included in the manifestlist.
+Existing architecture specific release payloads continue to be delivered through their individual release controllers. A new heterogeneous
+release controller is bootstrapped and displays "4-stable" entries for ART constructed release payloads. `oc adm release ...` verbs would be
+[enhanced][WRKLDS-370] to allow the creation and management of manifestlist release payloads. Sub-verbs like `mirror`, `info`, and `extract`
+must be adapted to support manifestlist based payloads. The trigger for creating the heterogeneous payloads will be whenever there is a new
+payload produced for one of the included architectures.
+
+In the case of `oc adm release new`, the manifestlist flow would be triggered whenever the CVO image in a source image/imagestream is a manifestlist.
+- If the source CVO image is a standard manifest, the generated release payload will also be a manifest.
+- If the source CVO image is a manifestlist, the generated release payload would be a manifestlist(containing a manifest for each architecture
+possessed by the CVO manifestlist).
+
+In either case, oc adm release new would permit non-CVO component images to be manifest or manifestlists and pass them through directly to the
+resultant release manifest(s). If a manifestlist release payload is generated, each architecture specific release payload manifest will reference
+the same pullspecs provided in the input imagestream.
+
+Since the internal Openshift registry does not support manifestlists, the release controller would use quay.io. There will be a new architecture option
+added to Cincinnati called "multi".
```

Comment:
This `Cincinnati` sentence seems like it should be shifted down to the next section.

Reply:
done.